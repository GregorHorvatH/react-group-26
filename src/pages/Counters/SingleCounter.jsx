import React, { useState, useEffect } from 'react';
import { createUseStyles } from 'react-jss';

const useStyles = createUseStyles({
  counter: {
    padding: 10,
  },
  buttons: {
    display: 'flex',
  },
  button: {
    '&:hover': {
      backgroundColor: 'red',
    },
  },
});

const limit = 500;

const SingleCounter = ({ initialValue = 0 }) => {
  const classes = useStyles();
  const [value, setValue] = useState(initialValue);
  const [step, setStep] = useState(1);

  const handleDecrement = () =>
    setValue((prevState) => (prevState - step < 0 ? 0 : prevState - step));

  const handleIncrement = () =>
    setValue((prevState) =>
      prevState + step > limit ? limit : prevState + step,
    );

  const handleChangeStep = (e) => setStep(Number(e.target.value));

  useEffect(() => {
    console.log('value:', value);
  }, [value]);

  return (
    <div className={classes.counter}>
      <h4>Single Counter</h4>

      <label>
        <span>step</span>
        <select value={step} onChange={handleChangeStep}>
          <option value="1">1</option>
          <option value="5">5</option>
          <option value="10">10</option>
          <option value="25">25</option>
          <option value="50">50</option>
        </select>
      </label>

      <div className={classes.buttons}>
        <button className={classes.button} onClick={handleDecrement}>
          -
        </button>
        <p>{value}</p>
        <button className={classes.button} onClick={handleIncrement}>
          +
        </button>
      </div>
      <hr />
    </div>
  );
};

export default SingleCounter;

// class MyComponent extends React.Component {
//   render() {
//     return (
//       <div className="my-component"></div>
//     );
//   }
// }

// MyComponent.propTypes = {};
// MyComponent.defaultProps = {};

// export default MyComponent;
