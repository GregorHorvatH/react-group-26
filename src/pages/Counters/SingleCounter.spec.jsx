import { render, waitFor } from '@testing-library/react';
import SingleCounter, { sum } from './SingleCounter';

describe('sum', () => {
  it('1 + 1 = 2', () => {
    expect(sum(1, 1)).toBe(2);
  });

  it('10 + 13 = 23', () => {
    expect(sum(10, 13)).toBe(23);
  });

  it('boooom', () => {
    expect(sum(100, 130)).toBe(230);
  });

  it('NaN check 1', () => {
    expect(sum(100)).not.toBeNaN();
    expect(sum()).not.toBeNaN();
    expect(sum()).toBe(0);
  });
});

describe('SingleCounter', () => {
  it('snapshot test', () => {
    const { container } = render(<SingleCounter value={333} />);

    expect(container).toMatchSnapshot();
  });

  it('value test', () => {
    const { getByTestId } = render(<SingleCounter value={333} />);

    expect(getByTestId('value').textContent).toBe('333');
  });

  it('check function call', () => {
    const onIncrement = jest.fn();
    const { getByTestId } = render(<SingleCounter onIncrement={onIncrement} />);

    getByTestId('increment-button').click();
    getByTestId('increment-button').click();
    getByTestId('increment-button').click();
    getByTestId('increment-button').click();
    getByTestId('increment-button').click();

    expect(onIncrement).toBeCalled();
    expect(onIncrement).toBeCalledTimes(5);
  });
});
