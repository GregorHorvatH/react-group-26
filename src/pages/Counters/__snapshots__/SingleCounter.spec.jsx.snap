// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleCounter snapshot test 1`] = `
<div>
  <div
    class="counter-0-2-1"
  >
    <h4>
      Single Counter
    </h4>
    <label>
      <span>
        step
      </span>
      <select>
        <option
          value="1"
        >
          1
        </option>
        <option
          value="5"
        >
          5
        </option>
        <option
          value="10"
        >
          10
        </option>
        <option
          value="25"
        >
          25
        </option>
        <option
          value="50"
        >
          50
        </option>
      </select>
    </label>
    <div
      class="buttons-0-2-2"
    >
      <button
        class="button-0-2-3"
      >
        -
      </button>
      <p
        data-testid="value"
      >
        333
      </p>
      <button
        class="button-0-2-3"
        data-testid="increment-button"
      >
        +
      </button>
    </div>
    <hr />
  </div>
</div>
`;
